core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -772,56 +802,81 @@ public Map<String, Object> getAdditionalAttachmentProperties(Attachment a) throw
 		return result;
 	}
 
+	public FolderId getFolderIdByFolderName(ExchangeService exchangeService, String folderName, boolean create, boolean skipCache) throws Exception {
+		ExchangeFileSystemResolver resolver = getResolver(folderName);
 
+		return getFolderIdByFolderName(exchangeService, resolver, create, skipCache);
+	}
 
-	public FolderId getFolderIdByFolderName(ExchangeService exchangeService, String folderName, boolean create) throws Exception{
-		FindFoldersResults findResults;
-		findResults = exchangeService.findFolders(basefolderId, new SearchFilter.IsEqualTo(FolderSchema.DisplayName, folderName), new FolderView(Integer.MAX_VALUE));
-		if (create && findResults.getTotalCount()==0) {
-			log.debug("creating folder [" + folderName + "]");
-			createFolder(folderName);
-			findResults = exchangeService.findFolders(basefolderId, new SearchFilter.IsEqualTo(FolderSchema.DisplayName, folderName), new FolderView(Integer.MAX_VALUE));
-		}
-		if (findResults.getTotalCount()==0) {
-			log.debug("folder [" + folderName + "] not found");
-			return null;
-		}
-		if (log.isDebugEnabled()) {
-			log.debug("amount of folders with name: " + folderName + " = " + findResults.getTotalCount());
-			log.debug("found folder with name: " + findResults.getFolders().get(0).getDisplayName());
+	private FolderId getFolderIdByFolderName(ExchangeService exchangeService, ExchangeFileSystemResolver resolver, boolean create, boolean skipCache) throws Exception {
+		FolderId folderId = skipCache ? null : cache.getFolderId(resolver);
+
+		if(folderId == null){
+			FindFoldersResults findResults;
+			FolderId basefolderId = cache.getBaseFolderId(resolver.getMailbox());
+			if(basefolderId==null){
+				basefolderId = getBaseFolderId(resolver.getMailbox(), getBaseFolder());
+			}
+
+			findResults = exchangeService.findFolders(basefolderId, new SearchFilter.IsEqualTo(FolderSchema.DisplayName, resolver.getFolderName()), new FolderView(Integer.MAX_VALUE));
```

Comment:
Zoekt deze exchangeService in de juiste mailbox?
Zou je hier niet (tijdelijk) de ANCHOR_HEADER moeten zetten?
Hoe weet de aanroepende partij dat de FolderId in z'n eigen of in een andere mailbox zit?

Hiervoor graag testen maken:
dat folderExists folders in de gevraagde mailbox vindt, en niet in z'n eigen.
dat een move naar die folder werkt.

Reply:
Ik heb aanpassingen gemaakt, is dit de juiste aanpak?

De testen ga ik morgen mee bezig.